packages/chain-adapters/src/utils/utxoUtils.ts

```diff
@@ -0,0 +1,64 @@
+import { BTCInputScriptType } from '@shapeshiftoss/hdwallet-core'
+import { Asset, ChainTypes } from '@shapeshiftoss/types'
+import { BIP32Params } from '@shapeshiftoss/types'
+import { ChainAdapter as BitcoinChainAdapter } from '../bitcoin/BitcoinChainAdapter'
+import { ChainAdapter as EthereumChainAdapter } from '../ethereum/EthereumChainAdapter'
+/**
+ * Utility function to bip32Params from script type
+ * Get return an object with bip32Params derived from script type and asset
+ * @param scriptType
+ * @param asset
+ * @returns BIP32Params
+ */
+export const bip32FromScript = (
+  scriptType: BTCInputScriptType,
+  asset: Asset
+): BIP32Params => {
+  const purpose = purposeFromScript(scriptType)
+
+  switch(asset.chain) {
+    case ChainTypes.Bitcoin: {
+      return { ...BitcoinChainAdapter.defaultBIP32Params, purpose }
+    }
+    case ChainTypes.Ethereum: {
+      return { ...EthereumChainAdapter.defaultBIP32Params, purpose }
+    }
+  }
+}
+
+/**
+ * Returns coin purpose from a given script type
+ * @param scriptType
+ * @returns 44 | 49 | 84
+ */
+export const purposeFromScript = (
```

Comment:
a purpose should not be derived from a script type; to do so assumes not only that every wallet follows BIP44, but that another BIP will always allocate a fresh purpose number to go with a new script type, and that's not guaranteed.

Reply:
what if I call it "bip44 purpose from script"?